src/libs/actions/PersistedRequests.ts

```diff
@@ -18,17 +18,49 @@ function clear() {
 }
 
 function save(requestToPersist: Request) {
-    const requests = [...persistedRequests];
-    const existingRequestIndex = requests.findIndex((request) => request.data?.idempotencyKey && request.data?.idempotencyKey === requestToPersist.data?.idempotencyKey);
-    if (existingRequestIndex > -1) {
-        // Merge the new request into the existing one, keeping its place in the queue
-        requests.splice(existingRequestIndex, 1, requestToPersist);
-    } else {
-        // If not, push the new request to the end of the queue
-        requests.push(requestToPersist);
+    const requests = [...persistedRequests, requestToPersist];
+    for (let i = requests.length - 1; i > 0; i--) {
+        // since we're deleting elements from the array, it may be reindexed.
+        // In this case, we need to make sure our index is still in bounds
+        if (i >= requests.length) {
+            // eslint-disable-next-line no-continue
+            continue;
+        }
+
+        const request = requests[i];
+
+        // identify and handle any existing requests that conflict with the new one
```

Comment:
Also just realized that we don't need to run this inside the loop. We only care about `getConflictingRequests`, etc... for the new request

Reply:
nice, this is considerably simpler now